pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/verification/VcVerificationServiceImpl.scala

```diff
@@ -0,0 +1,57 @@
+package io.iohk.atala.pollux.core.service.verification
+
+import zio.{IO, *}
+
+class VcVerificationServiceImpl extends VcVerificationService {
+  override def verify(
+      vcVerificationRequests: List[VcVerificationRequest]
+  ): IO[VcVerificationServiceError, List[VcVerificationResult]] = {
+    vcVerificationRequests.map(vcVerificationRequest =>
+      val verificationOutcomesZIO = ZIO.collectAll(
+        vcVerificationRequest.verifications
+          .map(verification => verify(vcVerificationRequest.credential, verification))
+      )
+
+      verificationOutcomesZIO.map(verificationOutcomes => {
```

Comment:
Thinking about it, this split between `success`, `failed`, `failedAsError`, and `failedAsWarning` does not seem necessary, given that all the details are available in the `VcVerification` object.

IMO, it will imply more refactoring and potential breaking changes in the API contract if, for instance, we decide to introduce more failure types in the future. I think we'd better rely on the consumers to make this split if they need to.